### PR TITLE
Implement utime/utimes/futimens

### PR DESCRIPTION
**Pull request by SonicMastr**

### CODE REVIEW

**Comment by isage** [comment]
Fix tabs/spaces mix, please.

**Comment by SonicMastr** [comment]
Oh ffs. Yeah. Sorry.

File: newlib/libc/sys/vita/sys/utime.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#ifndef _SYS_UTIME_H
```
Comment (d3m3vilurr) [inline newlib/libc/sys/vita/sys/utime.h:20]: please add license holder of this header

File: newlib/libc/sys/vita/utime.c [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <psp2/io/stat.h> 
+#include <psp2/rtc.h>
+
+#include "vitaerror.h"
+#include "vitadescriptor.h"
+#include "vitafs.h"
+
+#define TRY(x) {rid=x; if (rid < 0) {errno = __vita_sce_errno_to_errno(rid, ERROR_GENERIC); return -1;}}
```
Comment (d3m3vilurr) [inline newlib/libc/sys/vita/utime.c]: `TRY` action needs `free(full_path);` in `utimes`.
so maybe better it define in `utimes` and `undef` after function body.
(or just use `goto` to exit the function in error case :))

and also can you make it to `do { ... } while (0);` form?

**Review by d3m3vilurr — CHANGES_REQUESTED** [review]

**Pull request merged**